src/sbd-cluster.c

```diff
@@ -261,10 +265,25 @@ notify_timer_cb(gpointer data)
 
 #endif
         case pcmk_cluster_corosync:
+#if SUPPORT_COROSYNC && CHECK_CPG_HANDLE
+            {
+                unsigned int local_nodeid;
+
+                /* there is no value for a cpg_handle that is defined
+                   as invalid and there is 32bit randomness in it
+                   so going ahead with whatever found in
+                   cluster.cpg_handle should be safe
+                 */
+                if (cpg_local_get(cluster.cpg_handle, &local_nodeid) == CS_OK) {
```

Comment:
CPG doesn't need to be connected (cluster_connect_cpg()) before calling cpg_local_get()?

cluster will be initialized to all 0's because it's static, so I don't think there will be any randomness. Which is good because otherwise static analysis might complain about using uninitialized memory :)

Reply:
It needs the connection but that code is on pacemaker-side in crm_cluster_connect.
Question is if it wouldn't be cleaner to have something like crm_cluster_check in pacemaker doing basically cpg_local_get or whatever seems to be useful for the cluster-type.

Initially there is gonna be a 0 but 0 is as well a valid cpg_handle. So checking for 0 isn't safe.
Haven't run coverity over it but using static-memory shouldn't result in complaints I guess.
The randomness comes from corosync which adds a random number to the cpg_handle.

Reply:
Ah, makes sense now :)